pkg/autodiscovery/autoconfig.go

```diff
@@ -563,13 +559,30 @@ func (ac *AutoConfig) resolveTemplateForService(tpl integration.Config, svc list
 	return resolvedConfig, nil
 }
 
-// GetLoadedConfigs returns configs loaded
-func (ac *AutoConfig) GetLoadedConfigs() map[string]integration.Config {
+// WithLoadedConfigs calls the given function with the map of all
+// loaded configs.  This is done with the config store locked, so
+// callers should perform minimal work within f.
+func (ac *AutoConfig) WithLoadedConfigs(f func(map[string]integration.Config)) {
 	if ac == nil || ac.store == nil {
 		log.Error("Autoconfig store not initialized")
-		return map[string]integration.Config{}
+		f(map[string]integration.Config{})
+		return
 	}
-	return ac.store.getLoadedConfigs()
+	ac.store.withLoadedConfigs(f)
+}
+
+// AllLoadedConfigs returns a slice of all loaded configs.  This slice
+// is freshly created and will not be modified after return.
+func (ac *AutoConfig) AllLoadedConfigs() []integration.Config {
```

Comment:
What about `GetAllLoadedConfigs` ? Or simply `GetLoadedConfigs` ?

I don't know what the `AllLoadedConfigs` function does just by looking at the name. WDYT ?

Reply:
I'm a little worried that such a name starts to sound like GetAllConfigs, which is an "active" function rather than an accessor.  Maybe "CurrentLoadedConfigs"?

Reply:
OK, I understand the idea now. Then why not simply `LoadedConfigs` ?

Reply:
Sounds good..